lib/ansible/modules/web_infrastructure/sentry_project.py

```diff
@@ -0,0 +1,174 @@
+    type: str
+    required: true
+
+  state:
+    description:
+      - The state of project.
+    type: str
+    required: true
+    choices: [ "present", "absent" ]
+
+  team:
+    description:
+      - The name of the team to create a new project for.
+    type: str
+    required: true
+
+  url:
+    description:
+      - Sentry URL.
+    type: str
+    required: false
+    default: sentry.io
+
+notes:
+  - Module supports check_mode, but it can't provide 100 percent guarantee that the specified slug hasn't been used before.
+author: "Mikhail Naletov (@okgolove)"
+'''
+
+EXAMPLES = '''
+# Create project "backend" in "example" organization and "senior" team
+- sentry_project:
+    api_token: 1234567890abcdwxyz
+    organization: example
+    project_name: backend
+    project_slug: backend
+    state: present
+    team: senior
+    url: sentry.example.com
+'''
+
+RETURN = ''' # '''
+
+import json
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils._text import to_native
+from ansible.module_utils.six.moves.urllib.parse import urlparse
+
+
+def main():
+    arg_spec = dict(
+        api_token=dict(type="str", required=True),
+        organization=dict(type="str", required=True),
+        project_name=dict(type="str", required=True),
+        project_slug=dict(type="str", required=True),
+        state=dict(type="str", required=True, choices=['present', 'absent']),
+        team=dict(type="str", required=True),
+        url=dict(type="str", default="sentry.io")
+    )
+
+    module = AnsibleModule(argument_spec=arg_spec, supports_check_mode=True)
+
+    api_token = module.params['api_token']
+    organization = module.params['organization']
+    project_name = module.params['project_name']
+    project_slug = module.params['project_slug']
+    state = module.params['state']
+    team = module.params['team']
+    url = urlparse(module.params['url'])
+
+    if not url.scheme:
+        url = "https://{0}".format(url.path)
+
+    def does_project_exist(project_name, organization, team, api_token):
+        response, info = fetch_url(
+            module, "{url}/api/0/teams/{organization}/{team}/projects/".format(
+                url=url, organization=organization, team=team),
+            headers={"Authorization": "Bearer {0}".format(api_token)},
+            method="GET")
+        if info["status"] != 200:
+            module.fail_json(msg="Unable to obtain a project list, response status: {0}, message: {1}".format(
+                info["status"], info.get("msg", "No message has been provided")))
+        body = json.loads(response.read())
+        return bool([x for x in body if x["name"] == project_name])
+
+    if state == 'present':
+        if does_project_exist(project_name, organization, team, api_token):
+            module.exit_json(changed=False)
+        if not module.check_mode:
```

Comment:
IMO, better to add this condition before if state == 'present'

Reply:
Could you explain it?
As for me, it looks less logical and clear.

Reply:
Just IMO, no problem